xcube/dsio.py

```diff
@@ -128,6 +145,57 @@ def find_dataset_io(format_name: str, modes: Iterable[str] = None, default: Data
     return default
 
 
+def guess_dataset_format(path: str) -> Optional[str]:
+    """
+    Guess a dataset format for a file system path or URL given by *path*.
+
+    :param path: A file system path or URL.
+    :return: The name of a dataset format guessed from *path*.
+    """
+    dataset_io_fitness_list = guess_dataset_ios(path)
+    if dataset_io_fitness_list:
+        return dataset_io_fitness_list[0][0].name
+    return None
+
+
+def guess_dataset_ios(path: str) -> List[Tuple[DatasetIO, float]]:
```

Comment:
Dzelge and I don't understand the purpose of fitness. Could you please elaborate?